# The Simple Recommender

In [1]:
import numpy as np 
import pandas as pd

In [2]:
read = pd.read_csv('/Users/nandrajog/Downloads/movies_metadata.csv')

/anaconda3/envs/C-vision/lib/python3.6/site-packages/IPython/core/interactiveshell.py:3058: DtypeWarning: Columns (10) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [3]:
read['vote_count']

0        5415.0
1        2413.0
2          92.0
3          34.0
4         173.0
          ...  
45461       1.0
45462       3.0
45463       6.0
45464       0.0
45465       0.0
Name: vote_count, Length: 45466, dtype: float64

The metric 

V = No. of votes generated by the movies 

M = No. of minimum votes required for a movie to qualify 

R = The mean rating of the movie 

C = The mean rating of the whole dataset 

In [4]:
# #Calculate the number of votes garnered by the 80th percentile movie
# m = df['vote_count'].quantile(0.80)
# m

The prerequisties

let's calculate the value of m 

In [5]:
m = read['vote_count'].quantile(0.80)

In [6]:
m

50.0

it turn out the value of m is 50.0

We can see the only 20% of the movies have gained more then 50 votes

Another prerequistie that we want movies with above 45 min runtime and shorter then 300 

In [7]:
q_movies = read[(read['runtime'] >=45) & ( read['runtime']<= 300)]

In [8]:
# q_movies = df[(df['runtime'] >= 45) & (df['runtime'] <= 300)]

In [9]:
# q_movies = q_movies[q_movies['vote_count'] >= m]

In [10]:
movies = q_movies[q_movies['vote_count']>=m]

In [11]:
movies.shape

(8963, 24)

In [12]:
type(movies['vote_count'])

pandas.core.series.Series

Calculating the Score

Calculate the C 

In [13]:
C = movies['vote_average'].mean()

In [14]:
C

6.351500613633828

In [15]:
# Function to compute the IMDB weighted rating for each movie
def weighted_rating(x, m=m, C=C):
    v = x['vote_count']
    R = x['vote_average']
    # Compute the weighted score
    return (v/(v+m) * R) + (m/(m+v) * C)

Now we see the average rating of the movie is 6.35 

In [16]:
movies['vote_average']

0        7.7
1        6.9
2        6.5
4        5.7
5        7.7
        ... 
45268    5.3
45339    5.2
45343    4.0
45348    3.4
45441    5.2
Name: vote_average, Length: 8963, dtype: float64

In [17]:
movies['score'] = movies.apply(weighted_rating, axis=1)


/anaconda3/envs/C-vision/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  """Entry point for launching an IPython kernel.


In [18]:
movies

adult                              belongs_to_collection    budget  \
0      False  {'id': 10194, 'name': 'Toy Story Collection', ...  30000000   
1      False                                                NaN  65000000   
2      False  {'id': 119050, 'name': 'Grumpy Old Men Collect...         0   
4      False  {'id': 96871, 'name': 'Father of the Bride Col...         0   
5      False                                                NaN  60000000   
...      ...                                                ...       ...   
45268  False                                                NaN   5402000   
45339  False                                                NaN         0   
45343  False  {'id': 52888, 'name': 'The Visitors Collection...  25868826   
45348  False                                                NaN         0   
45441  False                                                NaN         0   

                                                  genres  \
0      [{'id': 16, 'name': 'Animation'}, {'id': 35, '...   
1      [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...   
2      [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...   
4                         [{'id': 35, 'name': 'Comedy'}]   
5      [{'id': 28, 'name': 'Action'}, {'id': 80, 'nam...   
...                                                  ...   
45268                     [{'id': 35, 'name': 'Comedy'}]   
45339                     [{'id': 35, 'name': 'Comedy'}]   
45343                     [{'id': 35, 'name': 'Comedy'}]   
45348  [{'id': 28, 'name': 'Action'}, {'id': 12, 'nam...   
45441                     [{'id': 35, 'name': 'Comedy'}]   

                                   homepage      id    imdb_id  \
0      http://toystory.disney.com/toy-story     862  tt0114709   
1                                       NaN    8844  tt0113497   
2                                       NaN   15602  tt0113228   
4                                       NaN   11862  tt0113041   
5                                       NaN     949  tt0113277   
...                                     ...     ...        ...   
45268                                   NaN  277839  tt4531694   
45339                                   NaN  430365  tt5804948   
45343                                   NaN  248705  tt2441982   
45348                                   NaN   44918  tt1640571   
45441                                   NaN   14008  tt0294425   

      original_language                original_title  \
0                    en                     Toy Story   
1                    en                       Jumanji   
2                    en              Grumpier Old Men   
4                    en   Father of the Bride Part II   
5                    en                          Heat   
...                 ...                           ...   
45268                fr                       Pattaya   
45339                fr                À bras ouverts   
45343                fr  Les Visiteurs: La Révolution   
45348                en                    Titanic II   
45441                en                   Cadet Kelly   

                                                overview  ...      revenue  \
0      Led by Woody, Andy's toys live happily in his ...  ...  373554033.0   
1      When siblings Judy and Peter discover an encha...  ...  262797249.0   
2      A family wedding reignites the ancient feud be...  ...          0.0   
4      Just when George Banks has recovered from his ...  ...   76578911.0   
5      Obsessive master thief, Neil McCauley leads a ...  ...  187436818.0   
...                                                  ...  ...          ...   
45268  Franky and Krimo dream of leaving the grey gri...  ...          0.0   
45339  Jean-Étienne Fougerole is an intellectual bohe...  ...          0.0   
45343  Stuck in the corridors of time, Godefroy de Mo...  ...          0.0   
45348  On the 100th anniversary of the original voyag...  ...          0.0   
45441  Hyperactive teenag

In [19]:
#Sort the movies 

In [21]:
movies = movies.sort_values('score', ascending=False)


In [25]:
movies[['title', 'vote_count', 'vote_average', 'score', 'runtime']].head(50)

title  vote_count  \
10309                    Dilwale Dulhania Le Jayenge       661.0   
314                         The Shawshank Redemption      8358.0   
834                                    The Godfather      6024.0   
40251                                     Your Name.      1030.0   
12481                                The Dark Knight     12269.0   
2843                                      Fight Club      9678.0   
292                                     Pulp Fiction      8670.0   
522                                 Schindler's List      4436.0   
23673                                       Whiplash      4376.0   
5481                                   Spirited Away      3968.0   
2211                               Life Is Beautiful      3643.0   
1178                          The Godfather: Part II      3418.0   
1152                 One Flew Over the Cuckoo's Nest      3001.0   
1176                                          Psycho      2405.0   
1184                     Once Upon a Time in America      1104.0   
351                                     Forrest Gump      8147.0   
1154                         The Empire Strikes Back      5998.0   
18465                               The Intouchables      5410.0   
289                           Leon: The Professional      4293.0   
3030                                  The Green Mile      4166.0   
1170                                      GoodFellas      3211.0   
2216                              American History X      3120.0   
1161                                    12 Angry Men      2130.0   
9698                            Howl's Moving Castle      2049.0   
2884                               Princess Mononoke      2041.0   
5878                                     City of God      1852.0   
877                                      Rear Window      1531.0   
5553                          Grave of the Fireflies       974.0   
1910                                   Seven Samurai       892.0   
1132                                 Cinema Paradiso       834.0   
15480                                      Inception     14075.0   
22879                                   Interstellar     11187.0   
7000   The Lord of the Rings: The Return of the King      8226.0   
256                                        Star Wars      6778.0   
46                                             Se7en      5915.0   
25465                                          Mommy       734.0   
586                         The Silence of the Lambs      4549.0   
4099                                         Memento      4168.0   
1213                                     The Shining      3890.0   
1057                                  Reservoir Dogs      3821.0   
49                                The Usual Suspects      3334.0   
32144                                           Room      2838.0   
1201                              Dead Poets Society      2786.0   
109                                      Taxi Driver      2632.0   
1159                  The Good, the Bad and the Ugly      2371.0   
5857                                     The Pianist      1927.0   
1138                                  Paths of Glory       565.0   
895                                 Sunset Boulevard       533.0   
1166                    Once Upon a Time in the West      1160.0   
3189                                     City Lights       444.0   

       vote_average     score  runtime  
10309           9.1  8.906716    190.0  
314             8.5  8.487223    142.0  
834             8.5  8.482314    175.0  
40251           8.5  8.400532    106.0  
12481           8.3  8.292091    152.0  
2843            8.3  8.289985    139.0  
292             8.3  8.288827    154.0  
522             8.3  8.278282    195.0  
23673           8.3  8.277988    105.0  
5481            8.3  8.275753    125.0  
2211            8.3  8.273619    116.0  
1178            8.3  8.271907    200.0  
1152            8.3  8.268068    133.0  
1176

# The knowledge-based recommender

In [95]:
import pandas as pd
import numpy as np 


In [96]:
read_data = pd.read_csv('/Users/nandrajog/Downloads/movies_metadata.csv',low_memory=False)

In [97]:
read_data.columns

Index(['adult', 'belongs_to_collection', 'budget', 'genres', 'homepage', 'id',
       'imdb_id', 'original_language', 'original_title', 'overview',
       'popularity', 'poster_path', 'production_companies',
       'production_countries', 'release_date', 'revenue', 'runtime',
       'spoken_languages', 'status', 'tagline', 'title', 'video',
       'vote_average', 'vote_count'],
      dtype='object')

“Ask the user for the genres of movies he/she is looking for

Ask the user for the duration

Ask the user for the timeline of the movies recommended

Using the information collected, recommend movies to the user that have a high weighted rating (according to the IMDB formula) and that satisfy the preceding conditions
”



In [98]:
read_data= read_data[['genres','runtime','title','release_date','vote_average','vote_count']]

In [99]:
read_data.head(5)

genres  runtime  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...     81.0   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...    104.0   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...    101.0   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...    127.0   
4                     [{'id': 35, 'name': 'Comedy'}]    106.0   

                         title release_date  vote_average  vote_count  
0                    Toy Story   1995-10-30           7.7      5415.0  
1                      Jumanji   1995-12-15           6.9      2413.0  
2             Grumpier Old Men   1995-12-22           6.5        92.0  
3            Waiting to Exhale   1995-12-22           6.1        34.0  
4  Father of the Bride Part II   1995-02-10           5.7       173.0

In [100]:
read_data['release_date'].isnull().sum()

87

There are 87 movies which don't have any release date so we need to convert those movies into pandas dataframe i.e NAN --> NAT

In [101]:
read_data['release_date'] = pd.to_datetime(read_data['release_date'],errors='coerce')

In [102]:
read_data['release_date']

0       1995-10-30
1       1995-12-15
2       1995-12-22
3       1995-12-22
4       1995-02-10
           ...    
45461          NaT
45462   2011-11-17
45463   2003-08-01
45464   1917-10-21
45465   2017-06-09
Name: release_date, Length: 45466, dtype: datetime64[ns]

In [103]:
read_data['year'] = read_data['release_date'].apply(lambda x: str(x).split('-')[0])

In [104]:
read_data['year']

0        1995
1        1995
2        1995
3        1995
4        1995
         ... 
45461     NaT
45462    2011
45463    2003
45464    1917
45465    2017
Name: year, Length: 45466, dtype: object

In [105]:
read_data.head(5)

genres  runtime  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...     81.0   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...    104.0   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...    101.0   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...    127.0   
4                     [{'id': 35, 'name': 'Comedy'}]    106.0   

                         title release_date  vote_average  vote_count  year  
0                    Toy Story   1995-10-30           7.7      5415.0  1995  
1                      Jumanji   1995-12-15           6.9      2413.0  1995  
2             Grumpier Old Men   1995-12-22           6.5        92.0  1995  
3            Waiting to Exhale   1995-12-22           6.1        34.0  1995  
4  Father of the Bride Part II   1995-02-10           5.7       173.0  1995

In [106]:
read_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 7 columns):
genres          45466 non-null object
runtime         45203 non-null float64
title           45460 non-null object
release_date    45376 non-null datetime64[ns]
vote_average    45460 non-null float64
vote_count      45460 non-null float64
year            45466 non-null object
dtypes: datetime64[ns](1), float64(3), object(3)
memory usage: 2.4+ MB


As we See 'release_date = object' now we def a function which will convert release_data into int type

In [107]:
def convert_int(x):
    try:
        return int(x)
    except:
        return 0

In [108]:
# int(read_data['release_date']. values)

In [109]:
read_data['year'] = read_data['year'].apply(convert_int)

In [110]:
read_data

genres  runtime  \
0      [{'id': 16, 'name': 'Animation'}, {'id': 35, '...     81.0   
1      [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...    104.0   
2      [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...    101.0   
3      [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...    127.0   
4                         [{'id': 35, 'name': 'Comedy'}]    106.0   
...                                                  ...      ...   
45461  [{'id': 18, 'name': 'Drama'}, {'id': 10751, 'n...     90.0   
45462                      [{'id': 18, 'name': 'Drama'}]    360.0   
45463  [{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...     90.0   
45464                                                 []     87.0   
45465                                                 []     75.0   

                             title release_date  vote_average  vote_count  \
0                        Toy Story   1995-10-30           7.7      5415.0   
1                          Jumanji   1995-12-15           6.9      2413.0   
2                 Grumpier Old Men   1995-12-22           6.5        92.0   
3                Waiting to Exhale   1995-12-22           6.1        34.0   
4      Father of the Bride Part II   1995-02-10           5.7       173.0   
...                            ...          ...           ...         ...   
45461                       Subdue          NaT           4.0         1.0   
45462          Century of Birthing   2011-11-17           9.0         3.0   
45463                     Betrayal   2003-08-01           3.8         6.0   
45464             Satan Triumphant   1917-10-21           0.0         0.0   
45465                     Queerama   2017-06-09           0.0         0.0   

       year  
0      1995  
1      1995  
2      1995  
3      1995  
4      1995  
...     ...  
45461     0  
45462  2011  
45463  2003  
45464  1917  
45465  2017  

[45466 rows x 7 columns]

In [111]:
read_data = read_data.drop('release_date',axis=1)

In [112]:
read_data

genres  runtime  \
0      [{'id': 16, 'name': 'Animation'}, {'id': 35, '...     81.0   
1      [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...    104.0   
2      [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...    101.0   
3      [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...    127.0   
4                         [{'id': 35, 'name': 'Comedy'}]    106.0   
...                                                  ...      ...   
45461  [{'id': 18, 'name': 'Drama'}, {'id': 10751, 'n...     90.0   
45462                      [{'id': 18, 'name': 'Drama'}]    360.0   
45463  [{'id': 28, 'name': 'Action'}, {'id': 18, 'nam...     90.0   
45464                                                 []     87.0   
45465                                                 []     75.0   

                             title  vote_average  vote_count  year  
0                        Toy Story           7.7      5415.0  1995  
1                          Jumanji           6.9      2413.0  1995  
2                 Grumpier Old Men           6.5        92.0  1995  
3                Waiting to Exhale           6.1        34.0  1995  
4      Father of the Bride Part II           5.7       173.0  1995  
...                            ...           ...         ...   ...  
45461                       Subdue           4.0         1.0     0  
45462          Century of Birthing           9.0         3.0  2011  
45463                     Betrayal           3.8         6.0  2003  
45464             Satan Triumphant           0.0         0.0  1917  
45465                     Queerama           0.0         0.0  2017  

[45466 rows x 6 columns]

In [113]:
read_data.head()

genres  runtime  \
0  [{'id': 16, 'name': 'Animation'}, {'id': 35, '...     81.0   
1  [{'id': 12, 'name': 'Adventure'}, {'id': 14, '...    104.0   
2  [{'id': 10749, 'name': 'Romance'}, {'id': 35, ...    101.0   
3  [{'id': 35, 'name': 'Comedy'}, {'id': 18, 'nam...    127.0   
4                     [{'id': 35, 'name': 'Comedy'}]    106.0   

                         title  vote_average  vote_count  year  
0                    Toy Story           7.7      5415.0  1995  
1                      Jumanji           6.9      2413.0  1995  
2             Grumpier Old Men           6.5        92.0  1995  
3            Waiting to Exhale           6.1        34.0  1995  
4  Father of the Bride Part II           5.7       173.0  1995

In [114]:
read_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 6 columns):
genres          45466 non-null object
runtime         45203 non-null float64
title           45460 non-null object
vote_average    45460 non-null float64
vote_count      45460 non-null float64
year            45466 non-null int64
dtypes: float64(3), int64(1), object(2)
memory usage: 2.1+ MB


In [115]:
read_data.iloc[0]  # First Row

genres          [{'id': 16, 'name': 'Animation'}, {'id': 35, '...
runtime                                                        81
title                                                   Toy Story
vote_average                                                  7.7
vote_count                                                   5415
year                                                         1995
Name: 0, dtype: object

In [116]:
read_data.iloc[0][0]

"[{'id': 16, 'name': 'Animation'}, {'id': 35, 'name': 'Comedy'}, {'id': 10751, 'name': 'Family'}]"

In [117]:
read_data['genres'] = read_data['genres'].fillna('[]')

In [121]:
read_data['genres'][0]

"[{'id': 16, 'name': 'Animation'}, {'id': 35, 'name': 'Comedy'}, {'id': 10751, 'name': 'Family'}]"

In [122]:
#Apply literal_eval to convert to the list object”

In [123]:
from ast import literal_eval

The above library convert str into list

In [124]:
read_data['genres'] = read_data['genres'].apply(literal_eval)

In [125]:
read_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 45466 entries, 0 to 45465
Data columns (total 6 columns):
genres          45466 non-null object
runtime         45203 non-null float64
title           45460 non-null object
vote_average    45460 non-null float64
vote_count      45460 non-null float64
year            45466 non-null int64
dtypes: float64(3), int64(1), object(2)
memory usage: 2.1+ MB


In [126]:
read_data['genres'] = read_data['genres'].apply(lambda x: [i['name']for i in x] )

In [127]:
read_data

genres  runtime                        title  \
0       [Animation, Comedy, Family]     81.0                    Toy Story   
1      [Adventure, Fantasy, Family]    104.0                      Jumanji   
2                 [Romance, Comedy]    101.0             Grumpier Old Men   
3          [Comedy, Drama, Romance]    127.0            Waiting to Exhale   
4                          [Comedy]    106.0  Father of the Bride Part II   
...                             ...      ...                          ...   
45461               [Drama, Family]     90.0                       Subdue   
45462                       [Drama]    360.0          Century of Birthing   
45463     [Action, Drama, Thriller]     90.0                     Betrayal   
45464                            []     87.0             Satan Triumphant   
45465                            []     75.0                     Queerama   

       vote_average  vote_count  year  
0               7.7      5415.0  1995  
1               6.9      2413.0  1995  
2               6.5        92.0  1995  
3               6.1        34.0  1995  
4               5.7       173.0  1995  
...             ...         ...   ...  
45461           4.0         1.0     0  
45462           9.0         3.0  2011  
45463           3.8         6.0  2003  
45464           0.0         0.0  1917  
45465           0.0         0.0  2017  

[45466 rows x 6 columns]

In [128]:
read_data.head()

genres  runtime                        title  \
0   [Animation, Comedy, Family]     81.0                    Toy Story   
1  [Adventure, Fantasy, Family]    104.0                      Jumanji   
2             [Romance, Comedy]    101.0             Grumpier Old Men   
3      [Comedy, Drama, Romance]    127.0            Waiting to Exhale   
4                      [Comedy]    106.0  Father of the Bride Part II   

   vote_average  vote_count  year  
0           7.7      5415.0  1995  
1           6.9      2413.0  1995  
2           6.5        92.0  1995  
3           6.1        34.0  1995  
4           5.7       173.0  1995

Here Now we are going to exploding genres 

In [129]:
s = read_data.apply(lambda x: pd.Series(x['genres']),axis=1).stack().reset_index(level=1,drop=True)

In [130]:
s

0        Animation
0           Comedy
0           Family
1        Adventure
1          Fantasy
           ...    
45461       Family
45462        Drama
45463       Action
45463        Drama
45463     Thriller
Length: 91106, dtype: object

In [131]:
s

0        Animation
0           Comedy
0           Family
1        Adventure
1          Fantasy
           ...    
45461       Family
45462        Drama
45463       Action
45463        Drama
45463     Thriller
Length: 91106, dtype: object

In [132]:
s.name = 'genre'

In [133]:
#Create a new dataframe gen_df which by dropping the old 'genres' feature and adding the new 'genre'.”

gen_df = read_data.drop('genres',axis=1).join(s)

In [134]:
gen_df.head()

runtime      title  vote_average  vote_count  year      genre
0     81.0  Toy Story           7.7      5415.0  1995  Animation
0     81.0  Toy Story           7.7      5415.0  1995     Comedy
0     81.0  Toy Story           7.7      5415.0  1995     Family
1    104.0    Jumanji           6.9      2413.0  1995  Adventure
1    104.0    Jumanji           6.9      2413.0  1995    Fantasy

In [135]:
gen_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 93548 entries, 0 to 45465
Data columns (total 6 columns):
runtime         93194 non-null float64
title           93527 non-null object
vote_average    93527 non-null float64
vote_count      93527 non-null float64
year            93548 non-null int64
genre           91106 non-null object
dtypes: float64(3), int64(1), object(2)
memory usage: 5.0+ MB


In [148]:
def build_chart(df,percetile):
    
     #Ask for preferred genres”
    print("Input a preferred genres") 
    genres = input()

    # Ask for lower limit of duration
    
    print("Input the lower duration")
    low_time = int(input())
    
    # Ask for upper limit of duration
    
    print("Input the Upper duration ")
    upper_limit = int(input())
    
    
    # Ask for the earliest year

    print("Input the earliest year")
    earliest_year = int(input())
    
    # Ask for the latest year
    print("Input the latest year ")
    latest_year = int(input())
    
    #Define a new movies variable to store the preferred movies. Copy the contents of gen_df to movies”
    
    movies = gen_df.copy()
    
    #filter movies on the basis of the above conditions
    
    movies = movies[(movies['genre'] == genres)& 
                    (movies['runtime'] >=low_time) & 
                    (movies['runtime']<= upper_limit) & 
                    (movies['year'] >= earliest_year) &
                    (movies['year']<= latest_year)]
    
    # Compute the value of C & M
    
    C = movies['vote_average'].mean()
    m = movies['vote_count'].quantile(percetile)
    
    #Only consider movies that have higher than m votes. Save this in a new dataframe q_movies”
    
    q_movies = movies.copy().loc[movies['vote_count'] >=m]
    
    q_movies['score'] = q_movies.apply(lambda x: (x['vote_count']/(x['vote_count']+m) * x['vote_average'])  + (m/(m+x['vote_count']) * C),axis=1)
#     print(q_movies)
    
    q_movies = q_movies.sort_values('score',ascending = False)
    
    return q_movies

#     return q_movies

    
    

In [147]:
build_chart(gen_df,percetile=0.9)

Input a preferred genres
Animation
Input the lower duration
30
Input the Upper duration 
120
Input the earliest year
2000
Input the latest year 
2006
       runtime                             title  vote_average  vote_count  \
3362      89.0             The Road to El Dorado           7.0       892.0   
3629      84.0                       Chicken Run           6.5      1190.0   
3891      78.0          The Emperor's New Groove           7.2      1544.0   
4178      90.0                             Shrek           7.3      4183.0   
4238      95.0         Atlantis: The Lost Empire           6.7      1257.0   
4756      92.0                    Monsters, Inc.           7.5      6150.0   
5084      81.0                           Ice Age           7.1      3954.0   
5255      83.0  Spirit: Stallion of the Cimarron           7.4       870.0   
5310      85.0                     Lilo & Stitch           7.1      1371.0   
5744      95.0                   Treasure Planet           7.2       9

runtime                             title  vote_average  vote_count  \
9698     119.0              Howl's Moving Castle           8.2      2049.0   
6232     100.0                      Finding Nemo           7.6      6292.0   
4756      92.0                    Monsters, Inc.           7.5      6150.0   
8234     115.0                   The Incredibles           7.4      5290.0   
4178      90.0                             Shrek           7.3      4183.0   
5084      81.0                           Ice Age           7.1      3954.0   
10385     77.0                      Corpse Bride           7.2      1957.0   
3891      78.0          The Emperor's New Groove           7.2      1544.0   
5255      83.0  Spirit: Stallion of the Cimarron           7.4       870.0   
5310      85.0                     Lilo & Stitch           7.1      1371.0   
5744      95.0                   Treasure Planet           7.2       980.0   
6740      85.0                      Brother Bear           6.9      1410.0   
3362      89.0             The Road to El Dorado           7.0       892.0   
7717      93.0                           Shrek 2           6.7      3082.0   
10979    117.0                              Cars           6.6      3991.0   
10089     86.0                        Madagascar           6.6      3322.0   
4238      95.0         Atlantis: The Lost Empire           6.7      1257.0   
10826     91.0             Ice Age: The Meltdown           6.5      3034.0   
3629      84.0                       Chicken Run           6.5      1190.0   
8238     100.0                 The Polar Express           6.4      1524.0   
10969     83.0                    Over the Hedge           6.3      1116.0   
11114     91.0                     Monster House           6.3       912.0   
9353      77.0                  The Lion King 1½           6.3       694.0   
11376     85.0                      Flushed Away           6.0       909.0   
9771      91.0                            Robots           6.0      1383.0   
11405    108.0                        Happy Feet           5.9      1457.0   
8180      90.0                        Shark Tale           5.8      1612.0   
10520     81.0                    Chicken Little           5.6       974.0   
7729      80.0                          Garfield           5.2       851.0   

       year      genre     score  
9698   2004  Animation  7.670715  
6232   2003  Animation  7.451616  
4756   2001  Animation  7.358604  
8234   2004  Animation  7.249783  
4178   2001  Animation  7.129783  
5084   2002  Animation  6.951051  
10385  2005  Animation  6.912591  
3891   2000  Animation  6.859424  
5255   2002  Animation  6.823784  
5310   2002  Animation  6.764251  
5744   2002  Animation  6.744307  
6740   2003  Animation  6.636115  
3362   2000  Animation  6.606068  
7717   2004  Animation  6.589654  
10979  2006  Animation  6.525804  
10089  2005  Animation  6.513427  
4238   2001  Animation  6.486148  
10826  2006  Animation  6.425227  
3629   2000  Animation  6.351831  
8238   2004  Animation  6.305320  
10969  2006  Animation  6.222072  
11114  2006  Animation  6.212139  
9353   2004  Animation  6.198286  
11376  2006  Animation  6.041281  
9771   2005  Animation  6.031836  
11405  2006  Animation  5.962831  
8180   2004  Animation  5.888529  
10520  2005  Animation  5.805335  
7729   2004  Animation  5.600643

In [149]:
# gen_df